std_execution.bs

```diff
@@ -3131,6 +3131,284 @@ template&lt;typename T>
+
+Insert a new subclause, Class template `in_place_stop_callback` <b>[stopcallback.inplace]</b>, after the section added above, as a new subclause of Stop tokens <b>[thread.stoptoken]</b>.
+
+#### General <b>[stopcallback.inplace.general]</b> #### {#spec-stopcallback.inplace.general}
+
+1.
+
+    <pre highlight=c++>
+    namespace std {
+      template&lt;class Callback>
+      class in_place_stop_callback {
+      public:
+        using callback_type = Callback;
+
+        // [stopcallback.inplace.cons], constructors and destructor
+        template&lt;class C>
+          explicit in_place_stop_callback(in_place_stop_token st, C&& cb)
+            noexcept(is_nothrow_constructible_v&lt;Callback, C>);
+        ~in_place_stop_callback();
+
+        in_place_stop_callback(const in_place_stop_callback&) = delete;
+        in_place_stop_callback(in_place_stop_callback&&) = delete;
+        in_place_stop_callback& operator=(const in_place_stop_callback&) = delete;
+        in_place_stop_callback& operator=(in_place_stop_callback&&) = delete;
+
+      private:
+        Callback callback;      // exposition only
+      };
+
+      template&lt;class Callback>
+        in_place_stop_callback(in_place_stop_token, Callback)
+          -> in_place_stop_callback&lt;Callback>;
+    }
+    </pre>
+
+2. *Mandates*: `in_place_stop_callback` is instantiated with an argument for the template parameter `Callback` that satisfies both `invocable` and `destructible`.
+
+3. *Preconditions*: `in_place_stop_callback` is instantiated with an argument for the template parameter `Callback` that models both `invocable` and `destructible`.
+
+4. *Recommended practice*: Implementation should use the storage of the `in_place_stop_callback` objects to store the state necessary for their association with an `in_place_stop_source` object.
+
+#### Constructors and destructor <b>[stopcallback.inplace.cons]</b> #### {#spec-stopcallback.inplace.cons}
+
+<pre highlight=c++>
+template<class C>
+explicit in_place_stop_callback(in_place_stop_token st, C&& cb)
+  noexcept(is_nothrow_constructible_v&lt;Callback, C>);
+</pre>
+
+1. *Constraints*: `Callback` and `C` satisfy `constructible_from<Callback, C>`.
+
+2. *Preconditions*: `Callback` and `C` model `constructible_from<Callback, C>`.
+
+3. *Effects*: Initializes `callback` with `std::forward<C>(cb)`.
+    If `st.stop_requested()` is `true`, then `std::forward<Callback>(callback)()` is evaluated in the current thread before the constructor returns.
+    Otherwise, if `st` has an associated `in_place_stop_source` object, registers the callback with the stop state of the `in_place_stop_source` that `st` is associated with such that `std::forward<Callback>(callback)()` is evaluated by the first call to `request_stop()` on an associated `in_place_stop_source`.
+    The `in_place_stop_callback` object being initialized becomes associated with the `in_place_stop_source` object that `st` is associated with, if any.
```

Comment:
```suggestion
3. *Effects*: Initializes `callback_` with `std::forward<C>(cb)`.
    If `st.stop_requested()` is `true`, then `std::forward<Callback>(callback_)()` is evaluated in the current thread before the constructor returns.
    Otherwise, if `st` has an associated `in_place_stop_source` object, registers the callback with the stop state of the `in_place_stop_source` that `st` is associated with such that `std::forward<Callback>(callback_)()` is evaluated by the first call to `request_stop()` on an associated `in_place_stop_source`.
    The `in_place_stop_callback` object being initialized becomes associated with the `in_place_stop_source` object that `st` is associated with, if any.
```